core/modules/layout/layout.module

```diff
@@ -1525,3 +1528,48 @@ function layout_set_handler_contexts(array &$contexts, array $all_contexts, arra
   // All required contexts populated successfully.
   return TRUE;
 }
+
+/**
+ * Implements hook_block_info().
+ */
+function layout_block_info() {
+  $blocks = array();
+  $blocks['custom_text_block'] = array(
+    'info' => t('Custom text'),
+    'description' => t('Create a custom text block'),
```

Comment:
Instead of using a hook-based block, maybe this should be a class-based one? It'd be good to have more examples of class-based blocks in core and I think that may be the direction we're headed in the future.